rust/hyperlane-base/src/metrics/custom_metrics.rs

```diff
@@ -108,12 +150,41 @@ impl AgentMetricsUpdater {
         }
     }
 
+    async fn update_block_details(&self) {
+        let block_height = self.chain_metrics.block_height.clone();
+        let gas_price = self.chain_metrics.gas_price.clone();
+        if let HyperlaneDomain::Unknown { .. } = self.conf.domain {
+            return;
+        };
+        let chain = self.conf.domain.name();
+        debug!("Updating metrics for chain ({:?})", chain);
+        let Ok(Some(current_block)) = self.provider.get_chain_metrics().await else {
+            return;
+        };
+
+        if let Some(block_height) = block_height {
+            let height = current_block.latest_block.number as i64;
+            trace!("Block height for chain {chain} is {height}");
+            block_height
+                .with(&hashmap! { "chain" => chain })
+                .set(height);
+        }
+        if let Some(gas_price) = gas_price {
+            let protocol = self.conf.domain.domain_protocol();
+            let decimals_scale = 10f64.powf(decimals_by_protocol(protocol).into());
+            let gas = u256_as_scaled_f64(current_block.gas_price, protocol) * decimals_scale;
+            trace!("Gas price for chain {chain} is {gas:.1} (lowest denomination)");
```

Comment:
nit: now that we have e.g. Neutron where we can have a gas price of 0.57 neutron, wonder if we should change the 0.1 here to 0.2